workers/generic-worker/d2g_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"runtime"
+	"testing"
+
+	"github.com/mcuadros/go-defaults"
+	"github.com/taskcluster/taskcluster/v54/tools/d2g/dockerworker"
+)
+
+func TestWithValidDockerWorkerPayload(t *testing.T) {
+	setup(t)
+	image := map[string]interface{}{
+		"name": "ubuntu:latest",
+		"type": "docker-image",
+	}
+	imageBytes, err := json.Marshal(image)
+	if err != nil {
+		t.Fatalf("Error marshaling JSON: %v", err)
+	}
+	payload := dockerworker.DockerWorkerPayload{
+		Command:    []string{"/bin/bash", "-c", "echo hello world"},
+		Image:      json.RawMessage(imageBytes),
+		MaxRunTime: 10,
+	}
+	defaults.SetDefaults(&payload)
+	td := testTask(t)
```

Comment:
where does the payload get set in the test? I see payload is set on line 23, but I don't see it used anywhere.

Reply:
In the `submitAndAssert()` helper :)

Reply:
lol